src/main/kotlin/org/rust/ide/inspections/fixes/AttachFileToModuleFix.kt

```diff
@@ -0,0 +1,187 @@
+
+/**
+ * Attaches a file to a Rust module.
+ *
+ * Before fix:
+ * foo.rs (not attached)
+ * lib.rs
+ *
+ * After fix:
+ * foo.rs
+ * lib.rs
+ *  mod foo;
+ */
+class AttachFileToModuleFix(file: RsFile,
+                            private val targetModuleName: String? = null) : LocalQuickFixOnPsiElement(file) {
+    override fun getFamilyName(): String = text
+    override fun getText(): String = "Attach file to ${targetModuleName ?: "a module"}"
+
+    override fun invoke(project: Project, file: PsiFile, startElement: PsiElement, endElement: PsiElement) {
+        val rsFile = startElement as? RsFile ?: return
+        val availableModules = findAvailableModulesForFile(project, rsFile)
+        if (availableModules.isEmpty()) return
+
+        if (availableModules.size == 1) {
+            insertFileToModule(rsFile, availableModules[0])
+        } else if (availableModules.size > 1) {
+            selectModule(rsFile, availableModules)?.let { insertFileToModule(rsFile, it) }
+        }
+    }
+
+    override fun startInWriteAction(): Boolean {
+        return false
+    }
+
+    companion object {
+        fun findAvailableModulesForFile(project: Project, file: RsFile): List<RsMod> {
+            val virtualFile = file.virtualFile ?: return emptyList()
+            val pkg = project.cargoProjects.findPackageForFile(virtualFile) ?: return emptyList()
+
+            val directory = virtualFile.parent ?: return emptyList()
+            val modules = mutableListOf<RsMod>()
+
+            if (file.isModuleFile) {
+                // package target roots in parent directory
+                for (target in pkg.targets) {
+                    val crateRoot = target.crateRoot ?: continue
+                    if (crateRoot.parent == directory.parent) {
+                        modules.addIfNotNull(crateRoot.toPsiFile(project)?.rustFile)
+                    }
+                }
+            } else {
+                // mod.rs in the same directory
+                modules.addIfNotNull(findModule(file, project, directory.findFileByRelativePath(RsConstants.MOD_RS_FILE)))
+
+                // module file in parent directory
+                if (pkg.edition == CargoWorkspace.Edition.EDITION_2018) {
+                    val parent = directory.parent
+                    modules.addIfNotNull(findModule(file, project, parent?.findFileByRelativePath("${directory.name}.rs")))
+                }
+
+                // package target roots in the same directory
+                for (target in pkg.targets) {
+                    val crateRoot = target.crateRoot ?: continue
+                    if (crateRoot.parent == directory) {
+                        modules.addIfNotNull(crateRoot.toPsiFile(project)?.rustFile)
+                    }
+                }
+            }
+
+            return modules
+        }
+    }
+}
+
+private fun selectModule(file: RsFile, availableModules: List<RsMod>): RsMod? {
+    if (isUnitTestMode) return MOCK!!(file, availableModules)
```

Comment:
Let's convert `MOCK!!` into `MOCK ?: error ("You should set mock module selector via `withMockModuleAttachSelector`")`
It may help to quickly understand why your test fail